lib/rubocop/cop/style/endless_method.rb

```diff
@@ -76,6 +141,14 @@ def handle_allow_style(node)
           end
         end
 
+        def handle_require_always_style(node)
+          return if node.endless? || !node.body || node.body.begin_type? || node.body.kwbegin_type?
+
+          add_offense(node, message: MSG_REQUIRE_ALWAYS) do |corrector|
```

Comment:
I see a possible problem if we report and autocorrect methods into endless methods that are too long for a single line. On the other hand, there's autocorrection in `Layout/LineLength` too, so some of these cases will be solved by the autocorrection loop.

Reply:
I wasn't aware of this problem. I will take a look too.

Reply:
```ruby
def handle_require_always_style(node)
  return if node.endless? || !node.body || node.body.begin_type? || node.body.kwbegin_type?

  line_length = node.body.source.length + definition.length
  return if line_length > config_store.line_length # or however this works

  add_offense(node, message: MSG_REQUIRE_ALWAYS) do |corrector|
    correct_to_endless_method(corrector, node)
  end
end

def definition(node) = "def #{node.method_name}#{arguments(node)} = "

def correct_to_endless_method(corrector, node)
  replacement = "#{definition}#{node.body.source}"
  corrector.replace(node, replacement)
end
```